nixos/modules/virtualisation/podman.nix

```diff
@@ -74,25 +82,50 @@ in
         The final Podman package (including extra packages).
       '';
     };
-
-
   };
 
-  config = lib.mkIf cfg.enable {
-
-    environment.systemPackages = [ cfg.package ]
-      ++ lib.optional cfg.dockerCompat dockerCompat;
-
-    environment.etc."cni/net.d/87-podman-bridge.conflist".source = utils.copyFile "${pkgs.podman-unwrapped.src}/cni/87-podman-bridge.conflist";
-
-    # Enable common /etc/containers configuration
-    virtualisation.containers.enable = true;
-
-    assertions = [{
-      assertion = cfg.dockerCompat -> !config.virtualisation.docker.enable;
-      message = "Option dockerCompat conflicts with docker";
-    }];
-
-  };
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      environment.systemPackages = [ cfg.package ]
+        ++ lib.optional cfg.dockerCompat dockerCompat;
+
+      environment.etc."cni/net.d/87-podman-bridge.conflist".source = utils.copyFile "${pkgs.podman-unwrapped.src}/cni/87-podman-bridge.conflist";
+
+      # Enable common /etc/containers configuration
+      virtualisation.containers = {
+        enable = true;
+        containersConf.extraConfig = lib.optionalString cfg.enableNvidia
+          (builtins.readFile toml.generate "podman-nvidia.toml" {
+            engine = {
+              conmon_env_vars = [ "PATH=${lib.makeBinPath [ pkgs.nvidia-docker ]}" ];
+              runtimes.nvidia = [ "${pkgs.nvidia-docker}/bin/nvidia-container-runtime" ];
+            };
+          });
+      };
 
+      assertions = [
+        {
+          assertion = cfg.dockerCompat -> !config.virtualisation.docker.enable;
+          message = "Option dockerCompat conflicts with docker";
+        }
+        {
+          assertion = cfg.enableNvidia -> !config.virtualisation.docker.enableNvidia;
+          message = "Option enableNvidia conflicts with docker.enableNvidia";
+        }
+      ];
+    }
+    (mkIf cfg.enableNvidia {
+      environment.etc."nvidia-container-runtime/config.toml".source =
```

Comment:
The docker version of this lives in the package and the docker module only needs to source it, I think we should to the same here.

Reply:
There are some values that are different between the podman version and docker version though: https://github.com/NVIDIA/nvidia-container-runtime/issues/85#issuecomment-604931556. How should we handle that?

Reply:
In particular `no-cgroups = true` for podman (rootless) and `no-cgroups = false` for docker and podman rootful.

Reply:
Create `pkgs/applications/virtualization/nvidia-docker/podman-config.toml` and source that here.

Reply:
The other issue is filling in the value of `glibc` which is done by filling in a the `installPhase` of `nvidia-docker`. That was the other reason I opted for the expression based version of this file.

Reply:
Reusing the `@@glibcbin@` `substituteInPlace` should be fine.